File: qc-storage/src/main/java/com/eager/questioncloud/storage/post/PostContentEntity.java
package com.eager.questioncloud.storage.post;

import com.eager.questioncloud.core.domain.post.vo.PostContent;
import com.eager.questioncloud.core.domain.post.vo.PostFile;
import com.eager.questioncloud.storage.post.converter.PostFileConverter;
import jakarta.persistence.Convert;
import jakarta.persistence.Embeddable;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Embeddable
@AllArgsConstructor
@NoArgsConstructor
@Getter
public class PostContentEntity {
    private String title;
    private String content;
    @Convert(converter = PostFileConverter.class)
    private List<PostFile> files;

    public static PostContentEntity from(PostContent postContent) {
        return new PostContentEntity(postContent.getTitle(), postContent.getTitle(), postContent.getFiles());
    }

    public PostContent toModel() {
        return new PostContent(title, content, files);
    }
}

File: qc-community/src/main/kotlin/com/eager/questioncloud/post/infrastructure/entity/PostContentEntity.kt
package com.eager.questioncloud.post.infrastructure.entity

import com.eager.questioncloud.post.domain.PostContent
import com.eager.questioncloud.post.domain.PostFile
import com.eager.questioncloud.post.infrastructure.converter.PostFileConverter
import jakarta.persistence.Convert
import jakarta.persistence.Embeddable

@Embeddable
class PostContentEntity private constructor(
    var title: String,
    var content: String,
    @Convert(converter = PostFileConverter::class)
    var files: List<PostFile>
) {
    fun toModel(): PostContent {
        return PostContent(title, content, files)
    }

    companion object {
        fun from(postContent: PostContent): PostContentEntity {
            return PostContentEntity(postContent.title, postContent.content, postContent.files)
        }
    }
}
